docs/further_useful_features.rst

```diff
@@ -0,0 +1,438 @@
+    ValueError: Input partitions for update have different tables than dataset:
+    Input partition tables: {'table2'}
+    Tables of existing dataset: ['table1', 'table2']
+
+
+Adding data to existing datasets is not the only functionality achievable within an update
+and it can even be used to remove or replace data. To do this we use the ``delete_scope``
+keyword argument as shown in the example below:
+
+.. ipython:: python
+
+    dm = update_dataset_from_dataframes(
+        None,
+        store=store_factory,
+        dataset_uuid="partitioned_dataset",
+        partition_on="E",
+        delete_scope=[{"E": "train"}]
+    )
+    dm.partitions
+
+
+As we can see, we specified using a dictionary that data where the column ``E`` has the
+value ``train`` should be removed. Looking at the partitions after the update, we see that
+the partition ``E=train`` has been removed.
+
+.. note:: We defined ``delete_scope`` over a value of ``E``. ``E`` also happens to be a
+    column that we partitioned by. This is because using ``delete_scope`` uses the same
+    underlying logic as the predicate-based filtering mentioned in :ref:`getting_started`.
+
+    Attempting to use ``delete_scope`` will *also* work on datasets not previously
+    partitioned on any column(s); in this case however the effect will simply be to remove
+    **all** previous partitions and replace them with the ones in the update.
+
+When  using ``delete_scope``, multiple values for the same column cannot be defined as a
+list but have to be specified instead as individual dictionaries, i.e.
+``[{"E": ["test", "train"]}]`` will not work but ``[{"E": "test"}, {"E": "train"}]`` will.
+
+.. ipython:: python
+
+    dm = update_dataset_from_dataframes(
+        None,
+        store=store_factory,
+        dataset_uuid="another_partitioned_dataset",
+        partition_on=["E", "F"],
+        delete_scope=[{"E": "train", "F": "foo"}, {"E": "test", "F": "bar"}],
+    )
+
+    dm.partitions  # `E=train/F=foo` and `E=test/F=bar` are deleted
+
+
+
+
+
+Garbage collection
+==================
+
+When ``kartothek`` is executing an operation, it makes sure to not
+commit changes to the dataset until the operation has been succesfully completed. If a
+write operation does not succeed for any reason, although there may be new files written
+to storage, those files will not used by the dataset as they will not be referenced in
+the ``kartothek`` metadata. Thus, when the user reads the dataset, no new data will
+appear in the output.
+
+Similarly, when deleting a partition, ``kartothek`` only removes the reference of that file
+from the metadata.
+
+
+These temporary files will remain in storage until a ``kartothek``  garbage collection
+function is called on the dataset.
+If a dataset is updated on a regular basis, it may be useful to run garbage collection
+periodically to decrease unnecessary storage use.
+
+An example of garbage collection is shown below. A file named ``trash.parquet`` is
```

Comment:
I think from a user perspective the `trash.parquet` is not necessarily intuitive. Think of using a partitioned dataset schema where the trash looks like `colA=1/label.parquet` and mention that this file might have been referenced before an update, delete or compaction operation